### PR TITLE
Mp bar segment

### LINKED ISSUE #1395: Make MP bar show how much the current organelle costs
- [ ] segmented MP bar to visually show how much your organelle would cost.

(https://community.revolutionarygamesstudio.com/upload://lnXDuVkQAlMZ9tyd0R1cUhrS5kX.jpeg)
image
 
- [ ] Allow us to de-select organelles

On the Community Forums:
https://community.revolutionarygamesstudio.com/t/editor-suggestions/2742

### PR DESCRIPTION
**Pull request by icedjuro**
Closes #1395

### CODE REVIEW

File: src/microbe_stage/editor/MicrobeEditor.cs [hunk 1]
```diff
@@ -1118,6 +1127,8 @@ private void RenderHighlightedOrganelle(int q, int r, int rotation)
         var toBePlacedOrganelle = SimulationParameters.Instance.GetOrganelleType(
             ActiveActionName);
 
+        OrganelleCost = toBePlacedOrganelle.MPCost;
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/MicrobeEditor.cs:1130]: This gets called a bunch of duplicate times.

I'd prefer if you put code like this in `UpdateEditor`

```
if (ActiveActionName != null && ShowHover)
{
OrganelleCost = ...
} else {
OrganelleCost = 0;
}
```

File: src/microbe_stage/editor/MicrobeEditor.cs [hunk 2]
```diff
@@ -14,6 +14,11 @@ public class MicrobeEditor : Node, ILoadableGameState
     /// </summary>
     public string NewName;
 
+    /// <summary>
+    ///   Cost of the organelle that is about to be placed
+    /// </summary>
+    public float OrganelleCost;
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/MicrobeEditor.cs:20]: Rename this to `CurrentOrganelleCost` to make it a bit more clearer what it is.

File: src/microbe_stage/editor/MicrobeEditorGUI.cs [hunk 3]
```diff
@@ -366,9 +371,31 @@ public void Init(MicrobeEditor editor)
...
+
+        if (possibleMutationPoints < 0)
+        {
+            mutationPointsSubtractBar.TextureProgress_ =
+                GD.Load<Texture>("res://assets/textures/gui/bevel/MpBarInvalid.png");
+        }
+        else
+        {
+            mutationPointsSubtractBar.TextureProgress_ =
+                GD.Load<Texture>("res://assets/textures/gui/bevel/MpBarSubtract.png");
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/MicrobeEditorGUI.cs:397]: GD.Load shouldn't be called each frame. Load in the _Ready method and store the object.

**Comment by icedjuro** [comment]
I also made sure the bar isn't segmented when the player is in freebuild.

**Pull request merged**